ext/xml/xml.c

```diff
@@ -1029,8 +1029,8 @@ static void php_xml_parser_create_impl(INTERNAL_FUNCTION_PARAMETERS, int ns_supp
 		} else if (strcasecmp(encoding_param, "US-ASCII") == 0) {
 			encoding = (XML_Char*)"US-ASCII";
 		} else {
-			php_error_docref(NULL, E_WARNING, "Unsupported source encoding \"%s\"", encoding_param);
-			RETURN_FALSE;
+			zend_argument_value_error(1, "is an unsupported source encoding");
```

Comment:
```suggestion
			zend_argument_value_error(1, "is not a supported source encoding");
```
...maybe? This reads nicer to me, but your version is also ok.